File: server/internal/dto/response/knowledge.go
// Package response 定义知识库管理相关的响应结构体。
package response

import "time"

// KBResponse 知识库响应（后台管理用）。
type KBResponse struct {
	ID              int64  `json:"id"`
	Name            string `json:"name"`
	Description     string `json:"description"`
	EmbeddingModel  string `json:"embedding_model"`
	VectorDimension int    `json:"vector_dimension"`
	LlmConfigID     int64  `json:"llm_config_id"`
	ArticleCount    int    `json:"article_count"`
	CreatedBy       int64  `json:"created_by"`
	CreatedAt       string `json:"created_at"`
	UpdatedAt       string `json:"updated_at"`
}

// KBListResponse 知识库列表响应。
type KBListResponse struct {
	Items []KBResponse `json:"items"`
}

// ArticleListResponse 文章列表响应（含分页）。
type ArticleListResponse struct {
	Articles []ArticleResponse `json:"articles"`
	Total    int64             `json:"total"`
}

// ArticleResponse 文章列表项响应。
type ArticleResponse struct {
	ID              int64     `json:"id"`
	KBID            int64     `json:"kb_id"`
	KBName          string    `json:"kb_name"`
	Title           string    `json:"title"`
	Content         string    `json:"content"`
	Category        string    `json:"category"`
	Tags            []string  `json:"tags"`
	Status          int16     `json:"status"`
	StatusText      string    `json:"status_text"`
	SourceType      int16     `json:"source_type"`
	SourceTypeText  string    `json:"source_type_text"`
	FileType        string    `json:"file_type"`
	MinioPath       string    `json:"minio_path"`
	WordCount       int       `json:"word_count"`
	ChunkCount      int       `json:"chunk_count"`
	ProcessStatus   string    `json:"process_status"`
	ProcessError    string    `json:"process_error"`
	CreatedBy       int64     `json:"created_by"`
	CreatedByName   string    `json:"created_by_name"`
	ReviewedBy      *int64    `json:"reviewed_by"`
	PublishedBy     *int64    `json:"published_by"`
	PublishedByName string    `json:"published_by_name"`
	ReviewComment   string    `json:"review_comment"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

// ArticleDetailResponse 文章详情响应（含切片）。
type ArticleDetailResponse struct {
	ArticleResponse
	Chunks []ChunkResponse `json:"chunks"`
}

// DocumentUploadResponse 文档上传响应（多文件）。
type DocumentUploadResponse struct {
	Documents []DocumentUploadItem `json:"documents"`
}

// DocumentUploadItem 单个文档上传响应项。
type DocumentUploadItem struct {
	ArticleID     int64  `json:"article_id"`
	FileName      string `json:"file_name"`
	FileSize      int64  `json:"file_size"`
	FileType      string `json:"file_type"`
	ProcessStatus string `json:"process_status"`
}

// DocumentStatusResponse 文档处理状态响应。
type DocumentStatusResponse struct {
	ArticleID     int64  `json:"article_id"`
	FileName      string `json:"file_name"`
	ProcessStatus string `json:"process_status"`
	ProcessError  string `json:"process_error"`
}

// ChunkResponse 知识切片响应。
type ChunkResponse struct {
	ID              int64     `json:"id"`
	KBID            int64     `json:"kb_id"`
	Content         string    `json:"content"`
	ChunkIndex      int       `json:"chunk_index"`
	EmbeddingModel  string    `json:"embedding_model"`
	VectorDimension int       `json:"vector_dimension"`
	CreatedAt       time.Time `json:"created_at"`
}

File: web/src/lib/api/knowledge.ts
import { apiFetch, apiFetchPage } from './client';
import { PAGE_SIZE } from './constants';

export interface KB { id: number; name: string; description: string; embedding_model: string; vector_dimension: number; llm_config_id: number; article_count: number; created_at: string; }
export interface Article { id: number; kb_id: number; kb_name: string; title: string; content: string; source_type: number; status: number; status_text: string; category: string; tags: string[]; word_count: number; chunk_count: number; file_type: string; process_status: string; process_error: string; created_by_name: string; created_at: string; updated_at: string; }
export interface ArticleDetail extends Article { chunks: unknown[]; reviewed_by: number | null; published_by: number | null; minio_path: string; }

// KB
export function getKBList() { return apiFetch<KB[]>('/api/v1/admin/knowledge-bases'); }
export function getPortalKBList() { return apiFetch<Pick<KB, 'id' | 'name' | 'description'>[]>('/api/v1/portal/knowledge-bases'); }
export function createKB(data: Record<string, unknown>) { return apiFetch<{ id: number }>('/api/v1/admin/knowledge-bases', { method: 'POST', body: JSON.stringify(data) }); }
export function updateKB(id: number, data: Record<string, unknown>) { return apiFetch<null>(`/api/v1/admin/knowledge-bases/${id}`, { method: 'PUT', body: JSON.stringify(data) }); }
export function deleteKB(id: number) { return apiFetch<null>(`/api/v1/admin/knowledge-bases/${id}`, { method: 'DELETE' }); }

// 文章
export function getArticleList(kbId: number, page: number, status?: string) {
  let url = `/api/v1/admin/knowledge-bases/${kbId}/articles?page=${page}&page_size=${PAGE_SIZE}`;
  if (status && status !== '-1') url += `&status=${status}`;
  return apiFetchPage<Article>(url);
}
export function getArticle(id: number) { return apiFetch<ArticleDetail>(`/api/v1/admin/articles/${id}`); }
export function createArticle(kbId: number, data: Record<string, unknown>) { return apiFetch<{ id: number }>(`/api/v1/admin/knowledge-bases/${kbId}/articles`, { method: 'POST', body: JSON.stringify(data) }); }
export function updateArticle(id: number, data: Record<string, unknown>) { return apiFetch<null>(`/api/v1/admin/articles/${id}`, { method: 'PUT', body: JSON.stringify(data) }); }
export function submitReview(id: number) { return apiFetch<null>(`/api/v1/admin/articles/${id}/submit-review`, { method: 'POST' }); }
export function reviewArticle(id: number, approved: boolean, review_comment?: string) { return apiFetch<null>(`/api/v1/admin/articles/${id}/review`, { method: 'POST', body: JSON.stringify({ approved, review_comment }) }); }
export function publishArticle(id: number) { return apiFetch<null>(`/api/v1/admin/articles/${id}/publish`, { method: 'POST' }); }
export function disableArticle(id: number) { return apiFetch<null>(`/api/v1/admin/articles/${id}/disable`, { method: 'POST' }); }
export function enableArticle(id: number) { return apiFetch<null>(`/api/v1/admin/articles/${id}/enable`, { method: 'POST' }); }

// 文档
export function uploadDocuments(kbId: number, files: FileList) {
  const fd = new FormData();
  Array.from(files).forEach((f) => fd.append('files', f));
  return apiFetch<{ documents: { article_id: number; file_name: string; process_status: string; process_error: string }[] }>(`/api/v1/admin/knowledge-bases/${kbId}/documents/upload`, { method: 'POST', body: fd, headers: {} as Record<string, string> });
}
